code/modules/recycling/disposal/bin.dm

```diff
@@ -331,32 +328,40 @@
 	..()
 	full_pressure = FALSE
 	pressure_charging = TRUE
-	update_icon()
+	update_appearance()
+
+/obj/machinery/disposal/bin/update_appearance(updates)
+	. = ..()
+	if((machine_stat & (BROKEN|NOPOWER)) || panel_open)
+		luminosity = 0
+		return
+	luminosity = 1
 
-/obj/machinery/disposal/bin/update_icon()
-	cut_overlays()
+/obj/machinery/disposal/bin/update_overlays()
+	. = ..()
 	if(machine_stat & BROKEN)
-		pressure_charging = FALSE
-		flush = FALSE
 		return
 
 	//flush handle
 	if(flush)
-		add_overlay("dispover-handle")
+		. += "dispover-handle"
 
 	//only handle is shown if no power
 	if(machine_stat & NOPOWER || panel_open)
 		return
 
 	//check for items in disposal - occupied light
 	if(contents.len > 0)
-		add_overlay("dispover-full")
+		. += "dispover-full"
+		. += emissive_appearance(icon, "dispover-full", alpha = src.alpha)
```

Comment:
This should update lum sources (See below too)

Reply:
addressed in 1389b29